### DISCUSSION TITLE
Kubectl: not found

### DISCUSSION THREAD

**Discussion by shintaii** [actions/actions-runner-controller#2121] [Questions] [answered]
I ust migrated from a VM running a Action Runner to ARC.

I have configured and installed using the yaml files (the Helm chart did not work). 

Workflows start fine, jobs are scheduled but a few steps in my workflow failes:

```
Run kubectl apply -f $GITHUB_WORKSPACE/k8s/env-configmap-staging.yaml -n alunno-staging && kubectl apply -f $GITHUB_WORKSPACE/k8s/secret-staging.yaml -n alunno-staging

/runner/_work/_temp/9b46015d-b661-49[7]
6fffd.sh: line 1: kubectl: command not found

Error: Process completed with exit code 127.
```

I did not change anything in my workflow files and it worked fine on my default configured action runner on a VM. Any ideas?

**Comment by kevholmes** [comment, marked as answer]
How are you installing kubectl into the runner container? What kind/version of ARC container are you using (dind, dind rootless etc) to run your Actions?

**Reply by shintaii** [reply]
Yeah stupid me, maybe should work less hours per day. there was no kubectl in the container... duh!

I made my own image:

```
FROM summerwind/actions-runner:latest

RUN sudo curl -LO https://dl.k8s.io/release/v1.26.0/bin/linux/amd64/kubectl && \
    sudo chmod +x ./kubectl && \
    sudo mv ./kubectl /usr/local/bin/kubectl
```

and done...

**Reply by kevholmes** [reply]
Hehe, well glad you got it sorted! 🥳

**Comment by trent-abc** [comment]
Another option for future readers, is to manually install each time, I got something similar to the following working for me:
```
name: get_pods

on:
  workflow_dispatch:

jobs:
  release:
    runs-on: mylabel-runnerdeploy
    steps:
      - uses: actions/checkout@v2
      - name: Get pods
        shell: bash
        run: |
          curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
          chmod +x ./kubectl
          curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
          unzip awscliv2.zip
          sudo ./aws/install
          KUBECONFIG=some_config.yaml ./kubectl get pods --namespace "some-namespace" --context "some-context"
```
